### PR TITLE
Troubleshooting: A drop-down editor does not show data

### PR DESCRIPTION
**Pull request by vladaskorohodova**

### CODE REVIEW

File: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/00 A Drop-Down Editor Does Not Show Data.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+This section describes the issues related to the following drop-down editors:
+
+- [SelectBox](/Documentation/Guide/UI_Components/SelectBox/Getting_Started_with_SelectBox/)
+- [Lookup](/Documentation/Guide/UI_Components/Lookup/Getting_Started_with_Lookup/)
+- [DropDownBox](/Documentation/Guide/UI_Components/DropDownBox/Getting_Started_with_DropDownBox/)
+- [TagBox](/Documentation/Guide/UI_Components/TagBox/Getting_Started_with_TagBox/)
+- [DataGrid](/Documentation/Guide/UI_Components/DataGrid/Getting_Started_with_DataGrid/)/[TreeList](/Documentation/Guide/UI_Components/TreeList/Getting_Started_with_TreeList/) lookup column editor
+
+Drop-down editors may not display data in different scenarios: on save, on first load, when the value property is changed in the code. They may also display different values, for example, IDs instead of text. Follow the steps below to learn how to troubleshoot your application. If the solutions listed here do not help, [create a ticket](https://supportcenter.devexpress.com/ticket/create) in our Support Center.
```
Comment (natakazakova) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/00 A Drop-Down Editor Does Not Show Data.md]: Let's motivate a customer to add details: 
```suggestion
Drop-down editors may not display data in different scenarios: on save, on first load, when the value property is changed in the code. They may also display different values, for example, IDs instead of text. Follow the steps below to learn how to troubleshoot your application. If the solutions listed here do not help, create a ticket in our Support Center and describe your issue in details: [Create a ticket](https://supportcenter.devexpress.com/ticket/create).
```

File: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/05 Key and ValueExpr Should Match.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
+If you specify a [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) for a drop-down editor, the [store](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/store/).[key](/Documentation/ApiReference/Data_Layer/ArrayStore/Configuration/#key) and the drop-down editor's [valueExpr](/Documentation/ApiReference/UI_Components/dxSelectBox/Configuration/#valueExpr) should be the same. In the examples below, the `ID` is used as both **key** and **valueExpr**. This rule applies both to standalone editors and DataGrid/TreeList lookup column editor.
```
Comment (natakazakova) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/05 Key and ValueExpr Should Match.md]: I am not sure about formatting. The guideline way is to use "`" for API member mentions, but make sure it will be consistent with the other text.
```suggestion
If you specify a [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) for a drop-down editor, the [store](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/store/).[key](/Documentation/ApiReference/Data_Layer/ArrayStore/Configuration/#key) value and the drop-down editor's [valueExpr](/Documentation/ApiReference/UI_Components/dxSelectBox/Configuration/#valueExpr) should be the same. In the examples below, the `ID` is used as both `key` and `valueExpr`. This rule applies both to standalone editors and DataGrid/TreeList lookup column editor.
```

File: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/10 DataField and ValueExpr Should Store The Same Keys.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
+To correctly substitute displayed values with required values in a DataGrid/TreeList lookup column, follow these steps:
+
+1. Define the [lookup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/).[dataSource](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/#dataSource).
+
+2. Specify the [lookup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/).[valueExpr](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/#valueExpr). It should match the [columns](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/).[dataField](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#dataField). Refer to the following topic for detailed explanation: [lookup - API Section](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/).
+
+3. Use the [displayExpr](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/#displayExpr) property to specify displayed values.
```
Comment (natakazakova) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/10 DataField and ValueExpr Should Store The Same Keys.md]: Try to use the strong verbs instead of "use". Not use, but set, define, specify, so a user can understand how exactly they can use this property without further explanation.

File: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/10 DataField and ValueExpr Should Store The Same Keys.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
+To correctly substitute displayed values with required values in a DataGrid/TreeList lookup column, follow these steps:
+
+1. Define the [lookup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/).[dataSource](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/#dataSource).
```
Comment (natakazakova) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/10 DataField and ValueExpr Should Store The Same Keys.md]:
```suggestion
1. Define the [lookup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/).[dataSource](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/#dataSource) property.
```

Comment (natakazakova) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/10 DataField and ValueExpr Should Store The Same Keys.md]:
```suggestion
1. Define the [lookup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/).[dataSource](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/#dataSource) property.
```

File: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/10 DataField and ValueExpr Should Store The Same Keys.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+2. Specify the [lookup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/).[valueExpr](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/#valueExpr). It should match the [columns](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/).[dataField](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#dataField). Refer to the following topic for detailed explanation: [lookup - API Section](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/).
```
Comment (natakazakova) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/10 DataField and ValueExpr Should Store The Same Keys.md]:
```suggestion
2. Specify the [lookup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/).[valueExpr](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/#valueExpr) property. Its value should match the [columns](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/).[dataField](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#dataField) property value. Refer to the following topic for detailed explanation: [lookup - API Section](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/).
```

File: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/10 DataField and ValueExpr Should Store The Same Keys.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+3. Use the [displayExpr](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/#displayExpr) property to specify displayed values.
+
+In the example below, `StateID` and `ID` store the same keys.
```
Comment (natakazakova) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/10 DataField and ValueExpr Should Store The Same Keys.md]:
```suggestion
The example below configures the DataGrid so `StateID` and `ID` store the same keys.
```

File: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/20 AspNet.Data CreateStore Processing.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
+If you use the [DevExtreme.AspNet.Data](https://github.com/DevExpress/DevExtreme.AspNet.Data#devextreme-aspnet-data).[createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) method to bind a standalone editor or DataGrid/TreeList lookup column, the editor expects all data operations (including value selection) to be performed on the server. Set the [createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) [loadMode](/Documentation/ApiReference/Data_Layer/CustomStore/Configuration/#loadMode) option to `raw` to automatically process data on a client-side.
```
Comment (natakazakova) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/20 AspNet.Data CreateStore Processing.md]:
```suggestion
If you call the [DevExtreme.AspNet.Data](https://github.com/DevExpress/DevExtreme.AspNet.Data#devextreme-aspnet-data).[createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) method to bind a standalone editor or DataGrid/TreeList lookup column, the editor expects all data operations (including value selection) to be performed on the server. Set the [createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) [loadMode](/Documentation/ApiReference/Data_Layer/CustomStore/Configuration/#loadMode) option to `raw` to process data on a client.
```

**Review by natakazakova — APPROVED** [review]

File: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/00 A Drop-Down Editor Does Not Show Data.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
+This section describes the issues related to the following drop-down editors:
+
+- [SelectBox](/Documentation/Guide/UI_Components/SelectBox/Getting_Started_with_SelectBox/)
+- [Lookup](/Documentation/Guide/UI_Components/Lookup/Getting_Started_with_Lookup/)
+- [DropDownBox](/Documentation/Guide/UI_Components/DropDownBox/Getting_Started_with_DropDownBox/)
+- [TagBox](/Documentation/Guide/UI_Components/TagBox/Getting_Started_with_TagBox/)
+- [DataGrid](/Documentation/Guide/UI_Components/DataGrid/Getting_Started_with_DataGrid/)/[TreeList](/Documentation/Guide/UI_Components/TreeList/Getting_Started_with_TreeList/) lookup column editor
```
Comment (artem-kurchenko) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/00 A Drop-Down Editor Does Not Show Data.md:7]: DataGrid and TreeList cannot be named as drop-down editors. I think it's more correct to write like 
[DataGrid|lookup](link_here)

File: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/10 DataField and ValueExpr Should Store The Same Keys.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
+To correctly substitute displayed values with required values in a DataGrid/TreeList lookup column, follow these steps:
+
+1. Define the [lookup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/).[dataSource](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/#dataSource) property.
+
+2. Specify the [lookup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/).[valueExpr](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/#valueExpr) property. Its value should match the [columns](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/).[dataField](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#dataField) property value. Refer to the following topic for detailed explanation: [lookup - API Section](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/).
```
Comment (artem-kurchenko) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/10 DataField and ValueExpr Should Store The Same Keys.md]: I would say `value` and `type` should be the same because "3" and 3 technically match if you compare them like 3 == "3". So typing is important because we compare them like 3 === "3" :) Our customers often consider this as matching values especially the problem appears when values loaded from the server via `CustomStore`

Comment (artem-kurchenko) [reply concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/10 DataField and ValueExpr Should Store The Same Keys.md]: UPD: nevermind, found it's in the next topic :D

File: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/20 AspNet.Data CreateStore Processing.md [hunk 10]
```diff
@@ -0,0 +1,6 @@
+If you call the [DevExtreme.AspNet.Data](https://github.com/DevExpress/DevExtreme.AspNet.Data#devextreme-aspnet-data).[createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) method to bind a standalone editor or DataGrid/TreeList lookup column, the editor expects all data operations (including value selection) to be performed on the server. Set the [createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) [loadMode](/Documentation/ApiReference/Data_Layer/CustomStore/Configuration/#loadMode) option to `raw` to process data on a client.
```
Comment (artem-kurchenko) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/20 AspNet.Data CreateStore Processing.md]: loadMode: "raw" won't be efficient if a drop-down editor has large data

I think we should also mention that:

In this case, unutilized our [AspNet.Data](https://github.com/DevExpress/DevExtreme.AspNet.Data#devextreme-aspnet-data) library to obtain correct data in the Controller: [API Controllers](https://docs.devexpress.com/AspNetCore/401020/devextreme-based-controls/concepts/bind-controls-to-data/api-controllers) for .NET-based applications. If you have another back-end type adjust your server logic to  return data in the proper format according to `loadOptions` values: [Server-Side_Data_Processing](https://js.devexpress.com/Documentation/Guide/Data_Binding/Specify_a_Data_Source/Custom_Data_Sources/#Load_Data/Server-Side_Data_Processing)

Only then mention about loadMode e.g.

if your backend doesn't support server operations, set loadMode....


I would also recommend adding a screenshot to demonstrate what we're talking about like I did  in the original topic. It makes text more clear because visual representation is quite important here as well.

File: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/00 A Drop-Down Editor Does Not Show Data.md [hunk 11]
```diff
@@ -0,0 +1,9 @@
...
+
+Drop-down editors may not display data in different scenarios: on save, on first load, when the value property is changed in the code. They may also display different values, for example, IDs instead of text. Follow the steps below to learn how to troubleshoot your application. If the solutions listed here do not help, create a ticket in our Support Center and describe your issue in details: [Create a ticket](https://supportcenter.devexpress.com/ticket/create).
```
Comment (Abadzhev) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/00 A Drop-Down Editor Does Not Show Data.md]: I'm confused about a couple of things. 

- "may not display data" - does this mean "display an empty edit box" or "display an empty dropdown list", or both, or maybe even something else? What I'm saying is "no data" is a big vague for a drop-down editor. Such an editor displays data in multiple places. We can be more specific about the issue. 
- "on save". I'm not sure what you imply. An editor displays data. Then I save some data changes (do they have to be changes in this editor or not?) and then the data disappears? Only during the save operation? Or data disappears forever? .... One can ask similar question about "when the value property is changed" or "on first load". I mean - does the data appear on subsequent loads? :-) In any event - I don't really see you reference those "scenarios" in the next sections. Maybe there's no reason to mention them here at all. Since the troubleshooting guide doesn't ask me to identify such a "scenario" - then why are we listing them here?

File: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/15 DataField and ValueExpr Should Be the Same Type.md [hunk 12]
```diff
@@ -0,0 +1,129 @@
+If you use a lookup column editor in the DataGrid or TreeList, you need to specify the [columns](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/).[dataField](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#dataField) property. The [store](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/store/).[key](/Documentation/ApiReference/Data_Layer/ArrayStore/Configuration/#key), the [lookup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/).[valueExpr](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/#valueExpr) and the [columns](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/).[dataField](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#dataField) values should be the same data type. In the code below, the `StateID` and `ID` fields should have the same type (for example, `String`).
```
Comment (Abadzhev) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/15 DataField and ValueExpr Should Be the Same Type.md]: We already stated above that `store.key` should match `lookup.valueExpr`. So we can remove `store.key` from the list here and simplify the description?

```suggestion
If you use a lookup column editor in the DataGrid or TreeList, you need to specify the [columns](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/).[dataField](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#dataField) property. The data type in that field must match the type of values in [lookup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/).[valueExpr](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/#valueExpr). In the code below, the `StateID` and `ID` fields should have the same type (for example, `String`).
```

File: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/20 AspNet.Data CreateStore Processing.md [hunk 13]
```diff
@@ -0,0 +1,12 @@
+If you call the [DevExtreme.AspNet.Data](https://github.com/DevExpress/DevExtreme.AspNet.Data#devextreme-aspnet-data).[createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) method to bind a standalone editor or DataGrid/TreeList lookup column, the editor expects all data operations (including value selection) to be performed on the server. Use the [DevExtreme.AspNet.Data](https://github.com/DevExpress/DevExtreme.AspNet.Data#devextreme-aspnet-data) library to obtain correct data in the [Controller](https://docs.devexpress.com/AspNetCore/401020/devextreme-based-controls/concepts/bind-controls-to-data/api-controllers) for .NET-based applications. If you have another back-end type, adjust your server logic to return data in the proper format according to the [loadOptions](/Documentation/ApiReference/Data_Layer/CustomStore/LoadOptions/) parameters. Refer to the following article for more information: [Server-Side Data Processing](Documentation/Guide/Data_Binding/Specify_a_Data_Source/Custom_Data_Sources/#Load_Data/Server-Side_Data_Processing).
```
Comment (Abadzhev) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/20 AspNet.Data CreateStore Processing.md]: "have" - is a vague/weak verb, so I changed to "work with"

"proper format according to loadOptions" - seems repetitive. If you just say "according to loadOptions", you would automatically imply some sort of a proper format. 

```suggestion
If you call the [DevExtreme.AspNet.Data](https://github.com/DevExpress/DevExtreme.AspNet.Data#devextreme-aspnet-data).[createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) method to bind a standalone editor or DataGrid/TreeList lookup column, the editor expects all data operations (including value selection) to be performed on the server. Use the [DevExtreme.AspNet.Data](https://github.com/DevExpress/DevExtreme.AspNet.Data#devextreme-aspnet-data) library to obtain correct data in the [Controller](https://docs.devexpress.com/AspNetCore/401020/devextreme-based-controls/concepts/bind-controls-to-data/api-controllers) for .NET-based applications. If you work with another back-end type, adjust your server logic to return data according to the [loadOptions](/Documentation/ApiReference/Data_Layer/CustomStore/LoadOptions/) parameters. Refer to the following article for more information: [Server-Side Data Processing](Documentation/Guide/Data_Binding/Specify_a_Data_Source/Custom_Data_Sources/#Load_Data/Server-Side_Data_Processing).
```

File: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/20 AspNet.Data CreateStore Processing.md [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+
+Examine your network requests to learn which [loadOptions](/Documentation/ApiReference/Data_Layer/CustomStore/LoadOptions/) parameters you need to specify. In the example below the parameter is `filter`.
+
+![DataGrid Lookup Network Request Parameters](/images/Troubleshooting/datagrid-lookup-network-request-parameters.png)
+
+if your back-end does not support server operations, set the [createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) [loadMode](/Documentation/ApiReference/Data_Layer/CustomStore/Configuration/#loadMode) option to `raw` to process data on a client.
```
Comment (Abadzhev) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/20 AspNet.Data CreateStore Processing.md]:
```suggestion
If your back-end does not support server operations, set the [createStore()](https://github.com/DevExpress/DevExtreme.AspNet.Data/blob/master/docs/client-side-with-jquery.md#api-reference) [loadMode](/Documentation/ApiReference/Data_Layer/CustomStore/Configuration/#loadMode) option to `raw` to process data on a client.
```

File: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/10 DataField and ValueExpr Should Store The Same Keys.md [hunk 15]
```diff
@@ -0,0 +1,98 @@
+To correctly substitute displayed values with required values in a DataGrid/TreeList lookup column, follow these steps:
```
Comment (Abadzhev) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/10 DataField and ValueExpr Should Store The Same Keys.md]: This part only marginally differs from the part that follows... We say that two fields must store the same values. Then we also emphasize that they should be of the same type. If they store the same values, then obviously the type is the same, right? 

Also - this section doesn't look like a part of a troubleshooting guide. It has more in common with a Getting Started. Shows me how to set up a lookup from scratch. 

Long story short - I think we can skip this entire step.

File: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/00 A Drop-Down Editor Does Not Show Data.md [hunk 16]
```diff
@@ -1,9 +1,9 @@
-This section describes the issues related to the following drop-down editors:
+This section describes the solutions to the issues (an empty edit box or an empty dropdown list) related to the following drop-down editors:
```
Comment (Abadzhev) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/00 A Drop-Down Editor Does Not Show Data.md]: I tried to incorporate issue descriptions instead of the word "issues". Not that parentheses are bad, but this is a pretty long and complex sentence. 

```suggestion
This section describes what you can do if the following drop-down editors display an empty edit box or an empty dropdown list:
```

File: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/00 A Drop-Down Editor Does Not Show Data.md [hunk 17]
```diff
@@ -0,0 +1,9 @@
+This section describes what you can do if the following drop-down editors display an empty edit box or an empty dropdown list:
```
Comment (dirk-pieterse) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/00 A Drop-Down Editor Does Not Show Data.md]:
```suggestion
This section describes what you can do if the following drop-down editors display an empty edit box or empty drop-down list:
```

File: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/00 A Drop-Down Editor Does Not Show Data.md [hunk 18]
```diff
@@ -0,0 +1,9 @@
...
+
+- [SelectBox](/Documentation/Guide/UI_Components/SelectBox/Getting_Started_with_SelectBox/)
+- [Lookup](/Documentation/Guide/UI_Components/Lookup/Getting_Started_with_Lookup/)
+- [DropDownBox](/Documentation/Guide/UI_Components/DropDownBox/Getting_Started_with_DropDownBox/)
+- [TagBox](/Documentation/Guide/UI_Components/TagBox/Getting_Started_with_TagBox/)
+- [DataGrid](/Documentation/Guide/UI_Components/DataGrid/Getting_Started_with_DataGrid/)/[TreeList](/Documentation/Guide/UI_Components/TreeList/Getting_Started_with_TreeList/) lookup column editor
+
+Follow the steps below to learn how to troubleshoot your application. If the solutions listed here do not help, create a ticket in our Support Center and describe your issue in details: [Create a ticket](https://supportcenter.devexpress.com/ticket/create).
```
Comment (dirk-pieterse) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/00 A Drop-Down Editor Does Not Show Data.md]:
```suggestion
Follow the steps below to troubleshoot your application. If the solutions listed here do not help, create a ticket in our Support Center and describe your issue in more detail: [Create a ticket](https://supportcenter.devexpress.com/ticket/create).
```

File: concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/05 Key and ValueExpr Should Match.md [hunk 19]
```diff
@@ -0,0 +1,100 @@
+If you specify a [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) for a drop-down editor, the [store](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/store/).[key](/Documentation/ApiReference/Data_Layer/ArrayStore/Configuration/#key) value and the drop-down editor's [valueExpr](/Documentation/ApiReference/UI_Components/dxSelectBox/Configuration/#valueExpr) should be the same. In the examples below, the `ID` is used as both `key` and `valueExpr`. This rule applies both to standalone editors and DataGrid/TreeList lookup column editor.
```
Comment (dirk-pieterse) [inline concepts/80 Troubleshooting/50 A Drop-Down Editor Does Not Show Data/05 Key and ValueExpr Should Match.md]:
```suggestion
If you specify a [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) for a drop-down editor, the [store](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/store/).[key](/Documentation/ApiReference/Data_Layer/ArrayStore/Configuration/#key) value and the drop-down editor's [valueExpr](/Documentation/ApiReference/UI_Components/dxSelectBox/Configuration/#valueExpr) should be the same. In the examples below, the `ID` is used as both `key` and `valueExpr`. This rule applies to standalone editors and DataGrid/TreeList lookup column editors.
```

**Review by dirk-pieterse — APPROVED** [review]
